lib/DelegatedModule.js

```diff
@@ -2,79 +2,80 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var Module = require("./Module");
-var OriginalSource = require("webpack-sources").OriginalSource;
-var RawSource = require("webpack-sources").RawSource;
-var WebpackMissingModule = require("./dependencies/WebpackMissingModule");
-var DelegatedSourceDependency = require("./dependencies/DelegatedSourceDependency");
+"use strict";
 
-function DelegatedModule(sourceRequest, data, type, userRequest) {
-	Module.call(this);
-	this.sourceRequest = sourceRequest;
-	this.request = data.id;
-	this.meta = data.meta;
-	this.type = type;
-	this.userRequest = userRequest;
-	this.built = false;
-	this.delegateData = data;
-}
-module.exports = DelegatedModule;
+const Module = require("./Module");
+const OriginalSource = require("webpack-sources").OriginalSource;
+const RawSource = require("webpack-sources").RawSource;
+const WebpackMissingModule = require("./dependencies/WebpackMissingModule");
+const DelegatedSourceDependency = require("./dependencies/DelegatedSourceDependency");
 
-DelegatedModule.prototype = Object.create(Module.prototype);
-DelegatedModule.prototype.constructor = DelegatedModule;
+class DelegatedModule {
+	constructor(sourceRequest, data, type, userRequest) {
+		Module.call(this);
```

Comment:
While `Module` has not yet been converted to `class` syntax, it is a class. _Never_ call a class as a function, they _will_ throw even if it's an ES5 construction pattern.

The DelegatedModule's prototype's prototype is also the `Module` prototype so this should be:

```js
class DelegatedModule extends Module {
  constructor(sourceRequest, data, type, userRequest) {
    super()
    /* ... */
  }
  /* ... */
}
```